### PR TITLE
Support workspace/xreferences

### PR DESCRIPTION
**Pull request by nicksnyder**

### CODE REVIEW

File: client/src/protocol.ts [hunk 1]
```diff
@@ -556,6 +557,10 @@ export interface ServerCapabilities {
 	 */
 	referencesProvider?: boolean;
 	/**
+	 * The server provides workspace references exporting support.
+	 */
+	xworkspaceReferencesProvider?: boolean;
```
Comment (rothfels) [inline client/src/protocol.ts:562]: workspaceXReferencesProvider or just xreferencesProvider seems more consistent.

File: client/src/tsconfig.json [hunk 2]
```diff
@@ -3,7 +3,7 @@
 		"target": "es6",
 		"module": "commonjs",
 		"moduleResolution": "node",
-		"sourceMap": false,
+		"sourceMap": true,
```
Comment (rothfels) [inline client/src/tsconfig.json]: Revert the changes in this file before merging (incl. the trailing newline).

File: client/package.json [hunk 3]
```diff
@@ -18,17 +18,16 @@
...
 	},
 	"scripts": {
-		"prepublish": "node ./node_modules/vscode/bin/install && tsc -p ./src",
+		"prepublish": "tsc -p ./src",
 		"compile": "tsc -p ./src",
 		"watch": "tsc -w -p ./src",
-		"update-vscode": "node ./node_modules/vscode/bin/install"
+		"update-vscode": ""
 	}
-}
```
Comment (rothfels) [inline client/package.json]: Remove newline (to reduce size of diff / be consistent w/ vscode style).

File: client/package.json [hunk 4]
```diff
@@ -18,17 +18,16 @@
...
 		"@types/node": "^6.0.42"
 	},
 	"dependencies": {
 		"@sourcegraph/vscode-jsonrpc": "^3.0.4-patch",
-		"vscode-languageserver-types": "^3.0.3"
+		"@sourcegraph/vscode-languageserver-types": "^3.0.3-patch"
 	},
 	"scripts": {
-		"prepublish": "node ./node_modules/vscode/bin/install && tsc -p ./src",
+		"prepublish": "tsc -p ./src",
```
Comment (rothfels) [inline client/package.json:28]: Are you going to commit this or revert it?

**Review by rothfels — APPROVED** [review]

File: client/src/protocol.ts (hunk 1 above)
Comment (nicksnyder) [reply client/src/protocol.ts:562]: This is the model for the json response so it needs to match the spec exactly:
https://github.com/sourcegraph/language-server-protocol/blob/master/extension-workspace-references.md

File: client/src/tsconfig.json (hunk 2 above)
Comment (nicksnyder) [reply client/src/tsconfig.json]: Is it bad to publish with source maps? What is the downside here?

File: client/package.json (hunk 4 above)
Comment (nicksnyder) [reply client/package.json:28]: It is impossible to compile this with vanilla VS code due to our proposed API changes. The only way to develop against this repo is to have these changes and have a local copy of vscode, so I was thinking I might as well check them in.

File: client/src/tsconfig.json (hunk 2 above)
Comment (rothfels) [reply client/src/tsconfig.json]: Not "bad", but uncommon for libraries to publish source maps. (Even if they're published, it's unlikely unless the sourcemaps are inlined that they would be picked up by your apps asset pipeline). And I doubt the vscode team would want this upstreamed.

File: client/src/protocol.ts (hunk 1 above)
Comment (rothfels) [reply client/src/protocol.ts:562]: Got it, makes sense.

File: client/src/tsconfig.json (hunk 2 above)
Comment (felixfbecker) [reply client/src/tsconfig.json]: @rothfels for TS libraries it's actually common to publish .js, .d.ts and .js.map so you can step into the library while debugging
